vortex-datafusion/src/persistent/opener.rs

```diff
@@ -294,17 +293,22 @@ impl FileOpener for VortexOpener {
                 );
             }
 
-            let filter = filter
-                .and_then(|f| {
-                    let exprs = split_conjunction(&f)
-                        .into_iter()
-                        .filter(|expr| can_be_pushed_down(expr, &predicate_file_schema))
-                        .collect::<Vec<_>>();
+            let filter = match filter {
+                None => None,
+                Some(f) => {
+                    let exprs = split_conjunction(&f).into_iter().collect::<Vec<_>>();
 
-                    make_vortex_predicate(&exprs).transpose()
-                })
-                .transpose()
-                .map_err(|e| DataFusionError::External(e.into()))?;
+                    for expr in &exprs {
+                        if !can_be_pushed_down(expr, &predicate_file_schema) {
+                            internal_datafusion_err!("DataFusion predicate {expr} cannot be pushed down to Vortex file {} with schema {predicate_file_schema}",
+                            file_meta.object_meta.location);
+                        }
```

Comment:
here be dragons when we have filters being applied to a column that doesn't exist in every file in the source.

if we have filters that touch columns which are not in the file's physical schema, we can't just skip it because the default value returned by the schema adapter might actually have failed the filter